modules/Bio/EnsEMBL/Production/Pipeline/Xrefs/Base.pm

```diff
@@ -63,18 +65,84 @@ sub create_db {
 }
 
 sub download_file {
-  my ($self, $file, $base_path, $source_name, $db, $skip_download_if_file_present, $release) = @_;
+  my ($self, $file, $base_path, $source_name, $db, $extra_args) = @_;
 
+  # Create uri object and get scheme
   my $uri = URI->new($file);
   if (!defined $uri->scheme) { return $file; }
+
+  # Get extra parameters
+  my $skip_download_if_file_present = $extra_args->{'skip_download_if_file_present'} || 0;
+  my $release = $extra_args->{'release'} || undef;
+  my $rel_number = $extra_args->{'rel_number'} || undef;
+  my $catalog = $extra_args->{'catalog'} || undef;
+
+  # Create file download path
   my $file_path;
   $source_name =~ s/\///g;
   my $dest_dir = catdir($base_path, $source_name);
   if (defined $db and $db eq 'checksum') {
     $dest_dir = catdir($base_path, 'Checksum');
   }
   make_path($dest_dir);
-  if ($uri->scheme eq 'ftp') {
+
+  # If file is in local ftp, copy from there
+  if ($file =~ /ftp.ebi.ac.uk/) {
+    # Construct local path
+    my $local_file = $file;
+    $local_file =~ s|http://ftp.ebi.ac.uk/pub/|/nfs/ftp/public/|;
```

Comment:
Should it match the `https` protocol instead?

Reply:
Hi Marc. Here we copy the files from our local ftp location rather than downloading, which was part of the change.